File: 2929. Distribute Candies Among Children II/sol.c
int max(int a, int b) {
    return a > b ? a : b;
}

int min(int a, int b) {
    return a < b ? a : b;
}

long long distributeCandies(int n, int limit) {
    long long res = 0;
    for(int i=0; i<=min(n, limit); i++) {
        res+=max(min(limit, n-i) - max(0,n-i-limit)+1, 0);
    }
    return res;
}

File: 3350. Adjacent Increasing Subarrays Detection II/sol.go
package main

func maxIncreasingSubarrays(nums []int) int {
	maxLen := 1
	prev := 0
	curr := 0
	for i := range nums {
		curr++
		if (i == len(nums)-1) || (nums[i] >= nums[i+1]) {
			maxLen = max(max(maxLen, curr/2), min(prev, curr))
			prev = curr
			curr = 0
		}
	}
	return maxLen
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}
